### PR TITLE
Worst season

### PR DESCRIPTION
**Pull request by gabichuelas**
@sciencefixion @rrabinovitch @evilaspaas1 
This PR includes a bunch of refactoring for `best_season`, and writing `beset_season` + ALL tests (including for helper methods).

### CODE REVIEW

**Review by sciencefixion — APPROVED** [review]

**Comment by sciencefixion** [comment]
Great work on the refactor!

**Comment by gabichuelas** [comment]
@sciencefixion @rrabinovitch @evilaspaas1 I pushed up to this branch again to fix the season integer/string problem. Seasons are now all strings in both our class files and test files for both mine and Dorion's methods. All tests pass.

Also, because I'd merged Ruthie's recent league PR, that broke the `find_team_by_id(id)` method because I'd changed it to `find_team_by(id)` for clarity and relationship to other helper methods. Can't remember specifics right now, but the TLDR is, everything works again and merging this PR will not break those methods.

**Review by sciencefixion — APPROVED** [review]
Thanks for all the work to better align us with the spec harness!

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@ def count_of_teams
     teams.count
   end
 
-  def find_team_by_id(id)
+  def find_team_by(id)
```
Comment (rrabinovitch) [inline lib/stat_tracker.rb:62]: I'm curious about the reasoning behind the renaming here?

**Review by rrabinovitch — APPROVED** [review]
sorry I keep commenting without getting an idea of the full picture first - trying to work on that :) but I now see your message about changing the `find_team_by(id)` method name!
great job w this refactoring and thanks again for catching the spec harness details

**Pull request merged**